client/src/main/scala/ch/epfl/bluebrain/nexus/iam/client/IamClient.scala

```diff
@@ -0,0 +1,137 @@
+package ch.epfl.bluebrain.nexus.iam.client
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.client.RequestBuilding.Get
+import akka.http.scaladsl.model.Uri.Query
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import akka.stream.ActorMaterializer
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient._
+import ch.epfl.bluebrain.nexus.commons.http.JsonLdCirceSupport._
+import ch.epfl.bluebrain.nexus.commons.http.{HttpClient, UnexpectedUnsuccessfulHttpResponse}
+import ch.epfl.bluebrain.nexus.commons.types.Err
+import ch.epfl.bluebrain.nexus.commons.types.HttpRejection.UnauthorizedAccess
+import ch.epfl.bluebrain.nexus.iam.client.config.IamClientConfig
+import ch.epfl.bluebrain.nexus.iam.client.types._
+import ch.epfl.bluebrain.nexus.rdf.Iri.Path
+import ch.epfl.bluebrain.nexus.rdf.Iri.Path._
+import ch.epfl.bluebrain.nexus.rdf.syntax.akka._
+import io.circe.generic.auto._
+import journal.Logger
+import monix.eval.Task
+
+import scala.concurrent.{ExecutionContext, Future}
+
+sealed trait IamClient[F[_]] {
+
```

Comment:
I think it would be also useful to have a method like `def authorize(path:Path, permission: Permission): F[Unit]`, which would just tell whether a user has a given permission on a path as that will be probably quite a fair bit of queries.
Another endpoint would probably be needed for that.